src/rabbit_channel.erl

```diff
@@ -1809,17 +1913,21 @@ notify_queues(State = #ch{consumer_mapping  = Consumers,
     {rabbit_amqqueue:notify_down_all(QPids, self(), Timeout),
      State#ch{state = closing}}.
 
-foreach_per_queue(_F, []) ->
-    ok;
-foreach_per_queue(F, [{_DTag, _CTag, {QPid, MsgId}}]) -> %% common case
-    F(QPid, [MsgId]);
+foreach_per_queue(_F, [], Acc) ->
```

Comment:
Can this function have two callbacks, or highlight the queue type more explicitly?

Reply:
We really can't be more explicit, unless we carry around the queue `type` apart from the `pid` field. If we store it as a tagged tuple or similar, it means to change all code that uses the `amqqueue` record.

Reply:
This is where we really need to start working on a good queue type abstraction to interact with rather than switching on the type the QPid.

Reply:
I just introduced some macros to distinguish between quorum/classic queues. It's only cosmetic, but might help: https://github.com/rabbitmq/rabbitmq-server/pull/1706/commits/80c85833bcab905c11c98b839b4cfb8bf1a2db27

`?IS_QUORUM(QPid)` and `?IS_CLASSIC(QPid)`